charge-holy-weapon.lic

```diff
@@ -38,27 +47,74 @@ class ChargeHolyWeapon
       message "1500+ charges remain, skipping..."
     end
 
-    @em.stow_weapon(@holy_weapon)
+    @em.stow_weapon(hw_settings['weapon_name'])
+  end
+
+  def complete_ritual(hw_settings)
+    weapon = hw_settings['weapon_name']
+    name = hw_settings['icon_name']
+    container = hw_settings['icon_container']
+    room = hw_settings['icon_room']
+
+    @em.stow_weapon(weapon)
+
+    unless icon
+      message "No icon found, using local chapel to complete the ritual!"
+      altar_room_ritual(weapon, @altar_room)
+    else
+      message "Icon -#{name}- found, moving to room #{room} to complete the ritual!"
+      icon_ritual(weapon, name, room, container)
+    end
   end
 
-  def complete_ritual
-    @em.stow_weapon(@holy_weapon)
-    walk_to(@altar_room)
+  def altar_room_ritual(weapon, room)
+    walk_to(room)
 
     case bput('pray chadatru', 'not cleared enough to pay proper respect to Chadatru', 'You decide to wait awhile longer')
     when 'not cleared enough to pay proper respect to Chadatru'
       message "Waiting for the prayer to finish (should take no longer than 2 minutes)..."
       waitfor('soothing sensation washes over your soul.')
 
-      @em.wield_weapon(@holy_weapon)
-      bput("glyph renew my #{@holy_weapon}", 'You trace')
-      @em.stow_weapon(@holy_weapon)
+      glyph_of_renew(weapon)
     when 'You decide to wait awhile longer'
       message "You're not ready to recharge yet, exiting!"
     end
 
     exit
   end
+
+  def icon_ritual(weapon, name, room = @altar_room, container)
+    walk_to(room)
+
+    bput('open my #{container}', 'You open')
+    case bput("get my #{name}", 'You get', 'What were')
```

Comment:
💡 If `container` was specified, is there any value to being more specific (e.g. `get my name from my container`)?